ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/AddressIndexedSignerProvider.java

```diff
@@ -12,66 +12,41 @@
  */
 package tech.pegasys.ethsigner.core;
 
-import tech.pegasys.signers.secp256k1.EthPublicKeyUtils;
 import tech.pegasys.signers.secp256k1.api.Signer;
 import tech.pegasys.signers.secp256k1.api.SignerProvider;
 
 import java.security.interfaces.ECPublicKey;
-import java.util.HashMap;
-import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
+import java.util.stream.Collectors;
 
-import org.apache.tuweni.bytes.Bytes;
-import org.web3j.crypto.Keys;
+import com.google.common.annotations.VisibleForTesting;
 
 public class AddressIndexedSignerProvider {
-
-  // Provides a signer based on its Public Key
   private final SignerProvider signerProvider;
 
-  // String is an 0x prefixed hex string of the ethereum address corresponding to the public
-  // key.
-  private final Map<String, ECPublicKey> addressToPublicKeyMap;
-
-  public AddressIndexedSignerProvider(
-      final SignerProvider signerProvider, final Map<String, ECPublicKey> addressToPublicKeyMap) {
+  public AddressIndexedSignerProvider(final SignerProvider signerProvider) {
     this.signerProvider = signerProvider;
-    this.addressToPublicKeyMap = addressToPublicKeyMap;
   }
 
   public static AddressIndexedSignerProvider create(final SignerProvider signerProvider) {
-    final Map<String, ECPublicKey> addrToPubKeyMap = new HashMap<>();
-
-    signerProvider
-        .availablePublicKeys()
-        .forEach(
-            pubKey -> {
-              final String address =
-                  "0x"
-                      + Keys.getAddress(
-                              Bytes.wrap(EthPublicKeyUtils.toByteArray(pubKey)).toHexString())
-                          .toLowerCase();
-              addrToPubKeyMap.put(address, pubKey);
-            });
-
-    return new AddressIndexedSignerProvider(signerProvider, addrToPubKeyMap);
+    return new AddressIndexedSignerProvider(signerProvider);
   }
 
   /* Gets a signer from its address, NOTE address MUST have 0x hex prefix */
```

Comment:
Is this still true? Think the handling of 0x in the Eth1AddressSignerIdentifier might make this work without the prefix?

Reply:
0x in address is not required anymore.